labeler-module/pkg/labeler/labeler.go

```diff
@@ -229,13 +239,69 @@ func (l *Labeler) getDriverLabelForNode(nodeName string) (string, error) {
 		}
 
 		if podutil.IsPodReady(pod) {
-			return "true", nil
+			return LabelValueTrue, nil
 		}
 	}
 
 	return "", nil
 }
 
+// getKataLabelForNode detects if Kata is enabled on the specified node
+// Returns "true" if Kata is enabled, "false" if not, or error if detection fails
+func (l *Labeler) getKataLabelForNode(nodeName string) (string, error) {
+	// Get or create cached detector for this node
+	cachedDetector, err := l.getOrCreateKataDetector(nodeName)
```

Comment:
We don't need an additional cache, the informers are already backed by a cache and we should be able to query the cache either using an indexer or a lister from the informer